### DISCUSSION TITLE
Host a PDS without a domain name on my PC for tinkering purposes

### DISCUSSION THREAD

**Discussion by harrmitch** [bluesky-social/atproto#1894] [How to do it?] [answered]
Is it possible to host the PDS without a domain name on a PC? This is for the purpose of tinkering and testing with the sandbox, for which buying a domain name and renting a server seem a bit excessive to me. Is it really possible at all, or what in the AT protocol would prevent a PDS to be hosted on a domain-less server, or my development computer?

**Comment by DavidBuchanan314** [comment]
I don't think a domain is necessary for a PDS, but you will need one if you want a handle that works properly. You should still be able to post without a valid handle, but the UI will show an "invalid handle" warning.

There are plenty of places that will give you free subdomains however - you can use github pages to get a `*.github.io` handle (using the HTTP verification method), or something like https://freedns.afraid.org/ (which gives you control of DNS records for a subdomain of your choice)

**Comment by bnewbold** [comment, marked as answer]
Seconding what David said: to have a fully interactive PDS instance in the network, it does need to be accessible at a regular DNS hostname, similar to email or web servers. You also need valid public DNS hostnames for individual user accounts. The short direct answer is that the Relay (aka BGS) needs to make a websocket connection to your PDS instance, via an HTTPS URL, to subscribe to repository updates from the PDS. The PDS is a server, not a client; you can't have the PDS "push" to the Relay.

You might be able to get away with a bunch of tinkering and experimentation locally. For example, our test suite runs things using local-only test domains. But various components may fail in unexpected ways, and this isn't something we "support".

There are a variety of options for getting a server on the web with a public DNS name, and they don't all cost money. For experimentation purposes it should be alright if the server is not online all the time. For example, you could use a tunneling or proxying system to expose a home server to the internet temporarily.
